tests/middlewares_tests/middlewares_test.go

```diff
@@ -0,0 +1,77 @@
+package middlewares_tests
+
+import (
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"testing"
+
+	"github.com/gin-gonic/gin"
+	"github.com/uc-cdis/cohort-middleware/middlewares"
+	"github.com/uc-cdis/cohort-middleware/tests"
+)
+
+func TestMain(m *testing.M) {
+	setupSuite()
+	retCode := m.Run()
+	tearDownSuite()
+	os.Exit(retCode)
+}
+
+func setupSuite() {
+	log.Println("setup for suite")
+}
+
+func tearDownSuite() {
+	log.Println("teardown for suite")
+}
+
+func setUp(t *testing.T) {
+	log.Println("setup for test")
+
+	// ensure tearDown is called when test "t" is done:
+	t.Cleanup(func() {
+		tearDown()
+	})
+}
+
+func tearDown() {
+	log.Println("teardown for test")
+}
+
+func TestPrepareNewArboristRequest(t *testing.T) {
+	setUp(t)
+	requestContext := new(gin.Context)
+	requestContext.Params = append(requestContext.Params, gin.Param{Key: "Authorization", Value: "dummy_token_value"})
+	requestContext.Writer = new(tests.CustomResponseWriter)
+	requestContext.Request = new(http.Request)
+	requestContext.Request.Header = map[string][]string{
+		"Authorization": {"dummy_token_value"},
+	}
+	u, _ := url.Parse("https://cohort-middleware/api/abc/123")
```

Comment:
should this be a relative path? Maybe yes...otherwise line 57 below is a bit strange...

Reply:
Yes, it should be relative, so in this case it'll use `/api/abc/123` for authentication, and resource will have prefix `/cohort-middleware` as an extra part.

Though, for deploying behind reverse proxy, it should use the same part, ignoring all the subpaths under which cohort-middleware is exposed, and, it currently works like that in [cloud-automation](https://github.com/uc-cdis/cloud-automation/blob/890867834f33a87cb58772f7203d9c8b95165b19/kube/services/revproxy/gen3.nginx.conf/cohort-middleware-service.conf#L5).

Reply:
Ok, the test seems correct to me. Thanks for the info.